### ISSUE TITLE
Can you talk about how you train, the steps of training (model source), thank you

### ISSUE THREAD

**Issue by abdulghafor** [herrfeder/AI_Cybersecurity_IDS_PoC#4]
hello，author。
Can you talk about how you train, the steps of training (model source), thank you

**Comment by herrfeder** [comment]
I was hoping, @MichaelSchwabe would give you some explanations as he did initially all the Modelling and Optimization stuff. I will head back to you until I reviewed it in-depth what Michael did :)

**Comment by MichaelSchwabe** [comment]
Hi @abdulghafor, 

I hope I help something. In the repo you will find the folder "analsys". There you have the folder "HP", which stands for Hyperparameter Optimization. Here you will find all a Jupyter notebook where everything can be run somewhat orderly and the code also *.py script. Here you get different models like a BaggingClassifier or RandomForest optimized on the used data. works also with other data, but they must be labeled because there are only supervised Learner in the pipeline. But take your time. Depending on the input it takes more than an hour until all models are optimized. All the metrics are also very clearly stored in a DataFrame.

The neural network was automatically selected and optimized with AutoKeras. Please take your time and patience for this. You can also find everything as JupyterNotebook or as *.py in the folder "NN".

We have chosen the methods of gridsearch for the naive procedures. We define a space of parameters and it runs through each model with every possible parameter constellation. The best models are selected based on their metrics. For example, we select the fastest ones (time to train), but also those that have a particularly high accuracy (also f1 score, precision, etc.).

The neural network is searched and optimized by AutoKeras Framework..

I hope I could help ... stay tuned


Summary:
Used Models in the Pipeline ... folder "HP"

my_models_name = [
    'RandomForestClassifier (Ensamble)',
    'BaggingClassifier (Ensamble)',
    'ExtraTreesClassifier (Ensamble)',
    'AdaBoostClassifier (Ensamble)',
    'GaussianNB (NaiveBayes)',
    'LogisticRegression (linear)'
]

**Comment by MichaelSchwabe** [comment]
i close this issue ... no more response from @abdulghafor
